src/frontend/Komponenter/Behandling/VedtakOgBeregning/Skolepenger/vedtaksvalidering.ts

```diff
@@ -0,0 +1,78 @@
+import { FormErrors } from '../../../../App/hooks/felles/useFormState';
+import { IUtgiftsperiodeSkolepenger } from '../../../../App/typer/vedtak';
+import { InnvilgeVedtakForm } from './VedtaksformSkolepenger';
+
+export const validerInnvilgetVedtakForm = ({
+    utgiftsperioder,
+    begrunnelse,
+}: InnvilgeVedtakForm): FormErrors<InnvilgeVedtakForm> => {
+    return {
+        ...validerPerioder({
+            utgiftsperioder,
+        }),
+        begrunnelse: !harVerdi(begrunnelse) ? 'Mangelfull utfylling av begrunnelse' : undefined,
+    };
+};
+
+export const validerPerioder = ({
+    utgiftsperioder,
+}: {
+    utgiftsperioder: IUtgiftsperiodeSkolepenger[];
+}): FormErrors<{
+    utgiftsperioder: IUtgiftsperiodeSkolepenger[];
+}> => {
+    return {
+        ...validerUtgiftsperioder({ utgiftsperioder }),
+    };
+};
+
+export const validerUtgiftsperioder = ({
+    utgiftsperioder,
+}: {
+    utgiftsperioder: IUtgiftsperiodeSkolepenger[];
+}): FormErrors<{ utgiftsperioder: IUtgiftsperiodeSkolepenger[] }> => {
+    const feilIUtgiftsperioder = utgiftsperioder.map((utgiftsperiode) => {
+        const { studietype, årMånedFra, årMånedTil, studiebelastning, utgifter } = utgiftsperiode;
+        const utgiftsperiodeFeil: FormErrors<IUtgiftsperiodeSkolepenger> = {
+            studietype: undefined,
+            årMånedFra: undefined,
+            årMånedTil: undefined,
+            studiebelastning: undefined,
+            utgifter: undefined,
+        };
+
+        if (!studietype) {
+            return { ...utgiftsperiodeFeil, studietype: 'Mangelfull utfylling av studietype' };
+        }
+        if (!årMånedFra || !årMånedTil) {
```

Comment:
Kanskje vi kan lage en egen oppgave på å legge inn valideringsregler for skolepenger? Her kunne man for eksempel lagt inn at årMånedFra må komme før årMånedEtter, etc.